libs/dao/src/main/java/com/akto/dao/testing/TestingRunDao.java

```diff
@@ -7,16 +7,33 @@
 import org.bson.types.ObjectId;
 
 import com.akto.dao.AccountsContextDao;
+import com.akto.dao.MCollection;
+import com.akto.dao.context.Context;
 import com.akto.dto.testing.TestingRun;
 import com.akto.dto.testing.TestingRunResultSummary;
 import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.CreateCollectionOptions;
 import com.mongodb.client.model.Filters;
 import com.mongodb.client.model.Projections;
 
 public class TestingRunDao extends AccountsContextDao<TestingRun> {
 
     public static final TestingRunDao instance = new TestingRunDao();
 
+    public void createIndicesIfAbsent() {
+        String dbName = Context.accountId.get()+"";
+        createCollectionIfAbsent(dbName, getCollName(), new CreateCollectionOptions());
+
+        String[] fieldNames = {TestingRun.SCHEDULE_TIMESTAMP};
+        MCollection.createIndexIfAbsent(getDBName(), getCollName(), fieldNames,false);
+
+        fieldNames = new String[]{TestingRun.END_TIMESTAMP};
+        MCollection.createIndexIfAbsent(getDBName(), getCollName(), fieldNames,false);
+
+        fieldNames = new String[]{"testingEndpoints.apiCollectionId", TestingRun.END_TIMESTAMP};
+        MCollection.createIndexIfAbsent(getDBName(), getCollName(), fieldNames,false);
+    }
```

Comment:
testing endpoints is an abstract field, due to which this index will only support one of its many implementations.